sdk/sdk.go

```diff
@@ -0,0 +1,403 @@
+// +build nra_enabled
+
+package sdk
+
+// #cgo LDFLAGS: -L/usr/local/lib -lnewrelic-collector-client -lnewrelic-common -lnewrelic-transaction
+// #include "newrelic_collector_client.h"
+// #include "newrelic_common.h"
+// #include "newrelic_transaction.h"
+// #include "stdlib.h"
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"unsafe"
+)
+
+// errNoMap is a map of errNo codes to error messages.
+var errNoMap = map[int]string{
+	-0x10001: "other",
+	-0x20001: "disabled",
+	-0x30001: "invalid param",
+	-0x30002: "invalid id",
+	-0x40001: "transaction not started",
+	-0x40002: "transaction in progress",
+	-0x40003: "transaction not named",
+}
+
+// errNo returns an error if the errno is < 0
+func errNo(i C.int, context string) (int, error) {
+	errno := int(i)
+	if errno < 0 {
+		errMsg := "unknown"
+		if e, ok := errNoMap[errno]; ok {
+			errMsg = e
+		}
+		return errno, errors.New(fmt.Sprintf("newrelic[%s]: %s", context, errMsg))
```

Comment:
If context is just the function name, we could potentially do it automatically:  http://stackoverflow.com/a/25927915 and save some typing